### PR TITLE
Milestone 3: Use Redux in React components

### PR DESCRIPTION
**Pull request by Kevin-Mena**
### In this Milestone, I Completed the following:
✅ Added Redux Toolkit
✅ Configured a Redux store
✅ Created slice states for books and categories

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import categoryReducer from './categories/categoriesSlice';
+
+const store = configureStore({
+  reducer: {
+    book: bookReducer,
+    category: categoryReducer,
+  },
+});
+
+export default store;
```
Comment (didierganthier) [inline src/redux/store.js:12]: Good job on the store  🏬 but the main React component should be wrapped with the `<Provider>` component from `react-redux` and the Redux `store` has been passed into the `<Provider>`, a little bit like that:
```
import React from "react"
import ReactDOM from "react-dom/client"
import { Provider } from "react-redux"
import store from "./redux/store"

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById("root")
)
```

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { v4 as uuid } from 'uuid';
+
+const initialState = {
+  books: [],
```
Comment (didierganthier) [inline src/redux/books/booksSlice.js:5]: This is good 💯 but try to add some books to your `initialState` to make sure that the books from that state are getting added. 🤓

**Review by didierganthier — CHANGES_REQUESTED** [review]
Hi @Kevin-Mena ,

🎉 Good job so far! You're making great progress on this project, but there are still a few areas that need attention before we can move on to the next step.



## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct Gitflow ✔️ 

## Required Changes 🛠
- [ ] You've done a great job 💯 but seems like you're not using the `store` you defined with Redux which makes you unable to fulfill some tasks:
   - the `books` should be be accessed from inside the `<BookList>` with `useSelector`


- Please review the comments under the review for specific changes that still need to be made.

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code.

🚀 Keep pushing forward, you're almost there! If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

**Please note**
- To avoid confusion and delays, please do not open a new pull request for re-reviews. Use the same pull request submitted for the first review, unless otherwise instructed.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Cheers and happy coding! 👏👏👏

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (Kevin-Mena) [reply src/redux/books/booksSlice.js:5]: Hello Reviewer, this pull request was for the previous milestone and I accidentally submitted it for the second milestone/exercise. Kindly give me any directions on how I will submit the next exercise/project

Comment (Kevin-Mena) [reply src/redux/books/booksSlice.js:5]: ### Dear reviewer 👋
This pull request was for the [Bookstore: add reducers and actions](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md) and accidentally submitted it for the [Bookstore: use Redux in React components.](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components_v2.md), kindly give me any guidance on how I will submit the next exercise,just reliased it after getting the first review from @didierganthier

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Kevin-Mena, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has the same PR for different milestone, I would implore you to use different PR for different milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (Kevin-Mena) [reply src/redux/books/booksSlice.js:5]: @Olamarx Noted👌

**Pull request merged**